2-Draft-Accepted/RFCXXXX-MSI-Installation-Path.md

```diff
@@ -0,0 +1,68 @@
+
+## Motivation
+
+Now that PowerShell Core 6.0 has GA'd, some users would like to run a preview version of PowerShell Core 6.1 side-by-side with PowerShell Core.
+Today, that's only possible if one version is installed via the MSI and another is installed via the "portable" ZIP package.
+Unfortunately, this makes the ZIP-installed copy of PowerShell difficult to install, and
+
+Also, in working to support Microsoft Update (MU) for servicing of PowerShell Core,
+we require that the path of all PowerShell Core binaries does not change between servicing.
+For example, if needed to patch a hypothetical version 6.0.3 to 6.0.4, the location of `pwsh.exe` and all associated binaries cannot change.
+
+Lastly, we now have multiple mechanisms for querying the version directly from the PowerShell binary itself.
+In order to support *nix-y semantics, we've added a `--version` parameter to `pwsh` that returns the version.
+We also provide the version of PowerShell in the FileVersion object associated with `pwsh`.
+
+## Specification
+
+* There shall be two "flavors" of MSIs available:
+    * Stable: a production-ready, supported build.
+    * Preview: an unsupported build that may contain incomplete new features and high-impact bugs,
+      and should therefore not be used in production.
+* All stable MSIs of PowerShell Core 6.x shall be installed to `$env:ProgramFiles\PowerShell\6\`.
+* All preview MSIs of PowerShell Core 6.x shall be installed to `$env:ProgramFiles\PowerShell\6-preview`.
+* When a stable MSI is run on a machine that already has a stable PowerShell Core 6.x installed,
+  an in-place update of that stable build shall be performed.
+* When a preview MSI is run on a machine that already has a preview PowerShell Core 6.x installed,
+  an in-place update of that stable build shall be performed.
+
+## Alternate Considerations and Proposals
+
+### Include the minor version number in the folder name
+
+This proposal would install/update all stable PowerShell Core 6.x builds to `$env:ProgramFiles\PowerShell\6.0`,
+and all stable PowerShell Core 6.1 builds to `$env:ProgramFiles\PowerShell\6.1`.
+Similarly, previews of PowerShell Core 6.1 would be installed to `$env:ProgramFiles\PowerShell\6.1-preview`,
+and previews of PowerShell Core 6.2 would be installed to `$env:ProgramFiles\PowerShell\6.2-preview`.
+
+This is undesirable given the Modern Lifecycle Policy that PowerShell Core has adopted.
+Essentially, we've given users 6 months to migrate off of an older minor version of PowerShell Core once a new one is released.
+Therefore, it makes more sense to *update* users from stable 6.0 to 6.1 than leave a version lying around that will soon become unsupported.
+If users want to maintain unsupported versions of PowerShell Core side-by-side,
+they should do a portable ZIP install.
+
+## Open Questions
+
+* What happens if `...\6\pwsh.exe` and `...\6-preview\pwsh.exe` are both on the PATH?
```

Comment:
Today, the msi installer automatically adds to the path.  Perhaps we should expose that as an Optional Action and have it checked by default with stable and unchecked by default for preview?

Reply:
Seems doable, but I still don't know what happens if I check it for both.

Reply:
Maybe not add preview to PATH at all. If the user want's the preview version in the PATH, PATH must be updated as a manual step (either machine wide or per user)? The user become responsible to making sure no duplicate path is in PATH. 
Alternative there could be a logic that check if any is on path, if not, add the preview to PATH. But I rather see this being an manual step regardless

Reply:
adding dynamic logic has a high failure rate.  

@SteveL-MSFT 's suggestion is more likely to be successful.

Reply:
>>What happens if `...\6\pwsh.exe` and `...\6-preview\pwsh.exe` are both on the PATH?
Former will be called, i.e. first one found while cycling through paths in the PATH.
Also, that means there is no sense to have both stable and preview on the PATH.